client/client.go

```diff
@@ -564,6 +559,11 @@ type (
 		SignalWithStartWorkflow(ctx context.Context, workflowID string, signalName string, signalArg interface{},
 			options StartWorkflowOptions, workflow interface{}, workflowArgs ...interface{}) (WorkflowRun, error)
 
+		// NewWithStartWorkflowOperation returns a WithStartWorkflowOperation to perform Update-with-Start.
+		// Returns an error if the WorkflowIDConflictPolicy is not set, or if the workflow or arguments are invalid.
+		// NOTE: Experimental
+		NewWithStartWorkflowOperation(options StartWorkflowOptions, workflow interface{}, args ...interface{}) (*WithStartWorkflowOperation, error)
```

Comment:
Yes, the SDK has a workflow name aliasing feature which uses a registry on the client instance: https://github.com/temporalio/sdk-go/blob/158b8231aef6bd73f4e632420899ad1b7f36cb89/internal/internal_worker.go#L1859

Reply:
Can't the workflow input be created as part of `UpdateWithStartWorkflow` instead of on instantiation of this? One might expect the actual call to do all of the validation instead of half here and half on the actual call. Arguably this should not return an error.

Reply:
After internal discussion, to allow users to provide their own implementation of this stateful thing, we are making the result an interface and keeping it on the client

Reply:
Now returns interface